homeassistant/components/sensor/sytadin.py

```diff
@@ -0,0 +1,130 @@
+"""
+Support for Sytadin Traffic, French Traffic Supervision.
+
+Systadin website : http://www.sytadin.fr
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sytadin/
+"""
+import logging
+
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.const import (LENGTH_KILOMETERS, CONF_NAME)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['lxml==4.0.0']
+
+SYSTADIN = 'http://www.sytadin.fr/sys/barometres_de_la_circulation.jsp.html'
+
+TRAFFIC_JAM_XPATH = '//*[@id="main_content"]/div[1]/div/span[3]/text()'
+MEAN_VELOCITY_XPATH = '//*[@id="main_content"]/div[2]/div/span[3]/text()'
+CONGESTION_XPATH = '//*[@id="main_content"]/div[3]/div/span[3]/text()'
+
+TRAFFIC_JAM_REGEX = '([0-9]+)'
+MEAN_VELOCITY_REGEX = '([0-9]+)'
+CONGESTION_REGEX = '([0-9]+.[0-9]+)'
+
+CONF_MONITORED_CONDITION = 'monitored_condition'
+OPTION_TRAFFIC_JAM = 'traffic_jam'
+OPTION_MEAN_VELOCITY = 'mean_velocity'
+OPTION_CONGESTION = 'congestion'
+CONF_UPDATE_INTERVAL = 'update_interval'
+
+SENSOR_TYPES = {
+    OPTION_TRAFFIC_JAM: ('Sytadin Traffic Jam', LENGTH_KILOMETERS),
+    OPTION_MEAN_VELOCITY: ('Sytadin Mean Velocity', LENGTH_KILOMETERS+'/h'),
+    OPTION_CONGESTION: ('Sytadin Congestion', '')
+}
+
+TIMEOUT = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITION): vol.In(SENSOR_TYPES.keys()),
```

Comment:
`.keys()` is not needed, it's the default.

Reply:
ok, fixed